### ISSUE TITLE
"Go backup in progress"-page saved in artifact that was fetched during GoCD-backup

### ISSUE THREAD

**Issue by Bjoern-Hegstam** [gocd/gocd#5221]
##### Issue Type

- Bug report

##### Summary

We encountered an issue where a task to fetch an artifact (a configuration file) from a previous stage was run at the same time as our GoCD server was backing up. The fetched file was still created, but instead of containing the expected configuration it contained the "Go backup in progress."-html page.

However, GoCD still ran our deploy step after this and we only noticed that something was wrong a while later when trying to load the deployed application.

I found this in our build log:
```plain
[go] Task: fetch artifact [src/config.js] => [src] from [app/build/build]took: 0.19s
[go] Fetching artifact [src/config.js] from [app/1572/build/latest/build]
[go] [WARN] The md5checksum value of the artifact [src/config.js] was not found on the server. Hence, Go could not verify the integrity of its contents.
[go] Saved artifact to [pipelines/app/src/config.js] without verifying the integrity of its contents.
[go] Task status: passed, took: 0.19s
```

I assume that the server backing up also resulted in the md5 checksum not being found. I read in the documentation that this is the intended behavior when a checksum could not be found, but it would be nice with an option to require checksums and fail the build otherwise. I was gonna submit that as a feature proposal, but the corrupted artifact seems odd enough that a bug report seems justified.

###### Basic environment details

* Go Version: `18.8.0 (7433-4bf750d09bf4dba76a2b7ce0d72e88ecdfdbd96a)`
* JAVA Version: `1.8.0_152`
* OS: `Linux 4.14.62-65.117.amzn1.x86_64`

**Comment by Bjoern-Hegstam** [comment]
I should also add that I have verified that the md5.checksum file is available in cruise-output and our jobs normally log that they have successfully verified the integrity of the artifact's contents

**Comment by arvindsv** [comment]
Agree. Both points make sense.

The first change I think should be made is to make that page return an HTTP status code of 500. Not 200.

**Comment by arvindsv** [comment]
Oh. This got closed automatically after the PR got merged. This should be fixed in the upcoming 18.10 release, @Bjoern-Hegstam. Thank you for bringing this up.

**Comment by Bjoern-Hegstam** [comment]
Awesome, thank you!
